src/main/java/org/springframework/data/r2dbc/repository/query/RowDataConverter.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.data.r2dbc.repository.query;
+
+import io.r2dbc.spi.Row;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.core.convert.converter.ConverterFactory;
+import org.springframework.data.r2dbc.repository.query.RowDataConverter.RowToNumberConverterFactory.RowToStringConverter;
+import org.springframework.util.Assert;
+import org.springframework.util.NumberUtils;
+
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * Base class for row data converter.
+ *
+ * @author Hebert Coelho
+ */
+abstract class RowDataConverter {
+    private RowDataConverter() {
+    }
+
+    static Collection<Object> getConvertersToRegister() {
+        List<Object> converters = new ArrayList<>();
+
+        converters.add(RowToBooleanConverter.INSTANCE);
+        converters.add(RowToDateConverter.INSTANCE);
+        converters.add(RowToInetAddressConverter.INSTANCE);
+        converters.add(RowToNumberConverterFactory.INSTANCE);
+        converters.add(RowToStringConverter.INSTANCE);
+        converters.add(RowToUuidConverter.INSTANCE);
+
+        return converters;
+    }
+
+    public enum RowToBooleanConverter implements Converter<Row, Boolean> {
+        INSTANCE;
+
+        @Override
+        public Boolean convert(Row row) {
+            return (Boolean) row.get(0);
```

Comment:
Use `row.get(0, Boolean.class)` to avoid downcasting.

This also applies for all other calls (except for the`RowToNumberConverterFactory`).

Reply:
awesome. I will do this refactorings and create a test for the converters.